frontend/src/components/ui/blockEditor/SpecsInterface.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useState } from 'react';
+import { TextInput, Button, Loading, TextArea } from '@carbon/react';
+import { pipelineAtom } from '@/atoms/pipelineAtom';
+import { useImmerAtom } from 'jotai-immer';
+import { Save } from "@carbon/icons-react";
+
+function SpecsInterface({ blockPath }) {
```

Comment:
I might be confused so maybe worth a discussion but why do we need to fetch a file from disk and save it to disk when the structure already exists in `pipeline.data`, which is automatically synced to disk? Can't we just mutate that structure using `useImmerAtom` like in the rest of the codebase?